File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/reserve/dtos/CompanyReserveSummary.java
package com.slotify.backend.spring.reserve.dtos;

import com.fasterxml.jackson.annotation.JsonTypeName;
import lombok.Getter;
import lombok.ToString;
import lombok.experimental.SuperBuilder;

import java.util.UUID;

@SuperBuilder
@ToString(callSuper = true)
@Getter
@JsonTypeName("COMPANY")
public class CompanyReserveSummary extends ReserveSummary {

    private UUID userId;

    private String userName;
    private String userImageUrl;

}

File: frontend/app/src/main/java/com/raj/slotify/dtos/reserves/CompanyReserveSummary.kt
package com.raj.slotify.dtos.reserves

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize
import java.time.LocalDateTime
import java.util.UUID

@Parcelize
data class CompanyReserveSummary(

    @SerializedName("reserveId")
    override val reserveId: UUID,

    @SerializedName("serviceId")
    override val serviceId: UUID,

    @SerializedName("startDateTime")
    override val startDateTime: LocalDateTime,

    @SerializedName("endDateTime")
    override val endDateTime: LocalDateTime,

    @SerializedName("minutesDuration")
    override val minutesDuration: Int,

    @SerializedName("serviceName")
    override val serviceName: String,

    @SerializedName("servicePriceCent")
    override val servicePriceCent: Int,

    @SerializedName("isCanceled")
    override val isCanceled: Boolean,

    @SerializedName("userId")
    val userId: UUID,

    @SerializedName("userName")
    val userName: String,

    @SerializedName("userImageUrl")
    val userImageUrl: String?,

): ReserveSummary (), Parcelable
